### ISSUE TITLE
[ntuple] Run DAOS unit tests if `-Dtesting=ON` was specified

### ISSUE THREAD

**Issue by jalopezg-r00t** [root-project/root#8074]
### Explain what you would like to see improved
RNTuple DAOS unit tests should be automatically run in a testing build.  Specifically, `-Ddaos_mock=ON` should be set if `testing` is defined.  `daos_mock` has no dependency on the real libdaos and is safe to run in our testing environment.

### Optional: share how it could be improved
Once PR #6825 is merged, a follow-up PR will be created to address this.
